source/extensions/common/wasm/wasm_extension.cc

```diff
@@ -58,9 +60,19 @@ WasmHandleExtensionFactory EnvoyWasm::wasmFactory() {
       // TODO(rapilado): Set the SanitizationConfig fields once sanitization is implemented.
       allowed_capabilities[capability.first] = proxy_wasm::SanitizationConfig();
     }
-    auto wasm = std::make_shared<Wasm>(vm_config.runtime(), vm_config.vm_id(),
-                                       anyToBytes(vm_config.configuration()), vm_key,
-                                       allowed_capabilities, scope, cluster_manager, dispatcher);
+    // clang-format off
+    std::unordered_map<std::string, std::string> envs;
+    // clang-format on
+    for (auto key : vm_config.env_keys()) {
+      auto value = std::getenv(key.data());
```

Comment:
While we expose those values as environment variables to Proxy-Wasm plugins, the source of them doesn't necessary have to be one, and I think we should allow control plane to set both: key and value, which I don't believe is possible with this PR.

Reading values from the Envoy's environment is probably fine, but I'm not sure how useful this is in practice.

Reply:
@mathetake thoughts on this?

Reply:
yeah, I now agree with @PiotrSikora here and the current api for just injecting Envoy's env to Wasm VMs is not that practical. We need to carefully design for enabling control planes to inject as well. 

That said, I would close this PR for now, and will put some design in https://github.com/envoyproxy/envoy/issues/14958